src/_pytest/capture.py

```diff
@@ -591,6 +589,11 @@ def snap(self):
         self.tmpfile.truncate()
         return res
 
+    def writeorg(self, data):
+        """ write to original file descriptor. """
+        data = data.encode("utf-8")  # XXX use encoding of original stream
```

Comment:
Hmmm not related to this code, but curious how `FDCaptureBinary` still creates `tmpfile` with `encoding="UTF-8"`...

Reply:
If you mean why `FDCaptureBinary` wraps the temporary file with `EncodedFile` (== `TextIOWrapper`), it's for two reasons:

1. In FD capturing, in addition to the FD redirection itself, it also does sys capturing. And the sys capturing requires a file object which looks like `sys.stdout` and friends, which are text-mode.
2. The `FDCapture` subclass uses it in its `snap()` implementation.

I have some plans to untangle all of this, but thought it'd be better not to mix refactorings with bug fixes.